src/ws/cockpitchannelresponse.c

```diff
@@ -211,11 +212,28 @@ cockpit_channel_response_close (CockpitChannelResponse *chesp,
   g_object_unref (chesp->transport);
   g_hash_table_unref (chesp->headers);
   cockpit_channel_inject_free (chesp->inject);
+  json_object_unref (chesp->open);
   g_free (chesp->channel);
   g_free (chesp);
 }
 
 static gboolean
+ensure_headers (CockpitChannelResponse *chesp,
```

Comment:
I think we also need a call to `ensure_headers` when a "done" message comes in.  Otherwise empty responses are broken.

```
diff --git a/src/ws/cockpitchannelresponse.c b/src/ws/cockpitchannelresponse.c
index 7f00217..21cdc40 100644
--- a/src/ws/cockpitchannelresponse.c
+++ b/src/ws/cockpitchannelresponse.c
@@ -354,6 +354,7 @@ on_transport_control (CockpitTransport *transport,
 
   if (g_str_equal (command, "done"))
     {
+      ensure_headers (chesp, 200, "OK");
       cockpit_web_response_complete (chesp->response);
       return TRUE;
     }
```